Makefile

```diff
@@ -19,140 +45,120 @@ SPHINXAUTOBUILD = "$(realpath bin/sphinx-autobuild)"
+# .PHONY: Test Performance
+# test-performance:
+# 	jmeter -n -t performance.jmx -l jmeter.jtl
 
-.PHONY: zpretty
-zpretty:  ## zpretty
-	if [ -f "bin/zpretty" ]; then zpretty -i ./**/*.zcml; fi
+# .PHONY: Test Performance Locust Querystring Search
+# test-performance-locust-querystring-search:
+# 	bin/locust -f performance/querystring-search.py --host http://localhost:12345/Plone --users 100 --spawn-rate 5 --run-time 5m --autostart
 
-.PHONY: python-clean
-python-clean:  ## Clean Python virtual environment
-	rm -rf bin include lib
+# .PHONY: Test Performance Locust Querystring Search CI
+# test-performance-locust-querystring-search-ci:
+# 	bin/locust -f performance/querystring-search.py --host http://localhost:12345/Plone --users 100 --spawn-rate 5 --run-time 5m --headless --csv=example
+
+## Documentation
 
 .PHONY: docs-clean
 docs-clean:  ## Clean current and legacy docs build directories
 	cd $(DOCS_DIR) && rm -rf $(BUILDDIR)/
 	rm -rf docs/build
 
-.PHONY: docs-html
-docs-html: bin/python  ## Build HTML
-	cd $(DOCS_DIR) && $(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html
-	@echo
-	@echo "Build finished. The HTML pages are in $(BUILDDIR)/html."
+$(BIN_FOLDER)/sphinx-autobuild $(BIN_FOLDER)/sphinx-build: $(BIN_FOLDER)/pip  ## Install dependencies for building docs
+	$(BIN_FOLDER)/pip install -r requirements-docs.txt
 
 .PHONY: docs-livehtml
-docs-livehtml: bin/python  ## Rebuild Sphinx documentation on changes, with live-reload in the browser
-	cd "$(DOCS_DIR)" && ${SPHINXAUTOBUILD} \
+docs-livehtml: $(BIN_FOLDER)/sphinx-autobuild  ## Rebuild Sphinx documentation on changes, with live-reload in the browser
+	cd "$(DOCS_DIR)" && $(BIN_FOLDER)/sphinx-autobuild \
 		--ignore "*.swp" \
 		-b html . "$(BUILDDIR)/html" $(SPHINXOPTS)
 
-.PHONY: docs-linkcheck
-docs-linkcheck: bin/python  ## Run linkcheck
-	cd $(DOCS_DIR) && $(SPHINXBUILD) -b linkcheck $(ALLSPHINXOPTS) $(BUILDDIR)/linkcheck
-	@echo
-	@echo "Link check complete; look for any errors in the above output " \
-		"or in $(BUILDDIR)/linkcheck/ ."
-
 .PHONY: docs-linkcheckbroken
-docs-linkcheckbroken: bin/python  ## Run linkcheck and show only broken links
-	cd $(DOCS_DIR) && $(SPHINXBUILD) -b linkcheck $(ALLSPHINXOPTS) $(BUILDDIR)/linkcheck | GREP_COLORS='0;31' grep -wi "broken\|redirect" --color=auto  && if test $$? = 0; then exit 1; fi || test $$? = 1
+docs-linkcheckbroken: $(BIN_FOLDER)/sphinx-build  ## Run linkcheck and show only broken links
+	cd $(DOCS_DIR) && $(BIN_FOLDER)/sphinx-build -b linkcheck $(ALLSPHINXOPTS) $(BUILDDIR)/linkcheck | GREP_COLORS='0;31' grep -wi "broken\|redirect" --color=auto  && if test $$? = 0; then exit 1; fi || test $$? = 1
 	@echo
 	@echo "Link check complete; look for any errors in the above output " \
 		"or in $(BUILDDIR)/linkcheck/ ."
 
-.PHONY: docs-vale
-docs-vale:  ## Run Vale style, grammar, and spell checks
-	vale sync
-	vale --no-wrap $(VALEFILES)
-	@echo
-	@echo "Vale is finished; look for any errors in the above output."
```

Comment:
Why was this removed? Codespell does not replace it.

Reply:
@stevepiercy I removed it because it was not functional before (it threw an error about .vale.ini missing) and was not referenced anywhere else. You are welcome to re-add it in a way that works.

Reply:
@stevepiercy (And sorry, I meant to ask you about it, but it got lost in the shuffle.)

Reply:
Vale v3 might have broken this. I'll dig into it. I kept it current with Volto and Documentation, where I use it most often.